### ISSUE TITLE
[APERO] mini data are not consistent

### ISSUE THREAD

**Issue by njcuk9999** [njcuk9999/apero-drs#708]

**Comment by clairem789** [comment]
here's a new link to Toulouse data for this minidataset: https://mycore.core-cloud.net/index.php/s/3cIsdql4lH3JqhV
thanks!

**Comment by njcuk9999** [comment]
So newworlds (as with previous versions) agrees very well with one group of the machines.

![image](https://github.com/njcuk9999/apero-drs/assets/6008608/d830e490-588d-498a-87dc-d88805c0c6ce)

And a zoom in:

![image](https://github.com/njcuk9999/apero-drs/assets/6008608/351e61bf-a791-4837-a3d8-41f4b26371d6)

rms of difference spirou@rali-cook@jupiter = 0.06417200529342021 m/s
rms of difference spirou@rali-cook@nb19 = 0.1346402795169976 m/s
rms of difference spirou@rali-spirou@maestria = 0.14235515258684286 m/s
rms of difference spirou@rali-newworlds = 0.13698797905443139 m/s
rms of difference spirou@rali-lam = 0.06417200529342021 m/s
rms of difference spirou@rali-cfht = 0.9038638383785134 m/s



So we have:
- cook@juptier, spirou@rali and lam agreeing very well
- cook@nb19, spirou@maestria and newworlds agreeing very well
- cfht not agreeing with anyone

**Comment by clairem789** [comment]
Interesting results! I wonder what this bimodality reflects. 
Can you please remind us on which machine the full reduction is run at UdM (rali, jupiter, maestria, nb19?)?

**Comment by njcuk9999** [comment]
The full run is on `rali`. And yep I'm equally interested/perplexed by this bimodality.

At one point @eartigau suggested it could be the CCF we do at the start in LBL "jumping" onto a certain solution for the sys vel but I'm not sure this could still be true.

I think the best way would be to have someone (with time) take a tar of the full mini data sets from all of these and work backward (start at the tcorr, then the extract, then the calibs, then pp) and see where the difference comes from.

If the tcorr are the same to ~numerical accuracy it has to come from LBL.

**Comment by larnoldgithub** [comment]
Neil I just sent you a link to a new set of blaze+tcorr from a new mini_run2 test I did yesterday night after a apero_reset and apero_precheck.

I plotted the difference between blaze or tcorr files from tests before the reset and after and I see differences, so maybe now the cfht data fits the results from other machines.

**Comment by njcuk9999** [comment]
Interesting, I'll add this data today. If it does agree with the other machines we really need to understand what happened between the restart. I'll post here when I have the data added to the graph.

**Comment by njcuk9999** [comment]
Yep so CFHT agrees well now. So the question is what was the difference? We might have to ask Chris what could have changed as that is really perplexing.

Here is the new graph:

![image](https://github.com/njcuk9999/apero-drs/assets/6008608/dfc893ab-e088-42d9-b395-d2757302e1da)


rms of difference spirou@rali-cook@jupiter = 0.06417200529342021 m/s
rms of difference spirou@rali-cook@nb19 = 0.1346402795169976 m/s
rms of difference spirou@rali-spirou@maestria = 0.14235515258684286 m/s
rms of difference spirou@rali-newworlds = 0.13698797905443139 m/s
rms of difference spirou@rali-lam = 0.06417200529342021 m/s
rms of difference spirou@rali-cfht = 0.06658835428246165 m/s


I'm glad there are now only two sets, I still don't understand that or why CFHT data changed after a reset! :man_shrugging:

**Comment by larnoldgithub** [comment]
![Screenshot 2023-08-13 at 21 36 01](https://github.com/njcuk9999/apero-drs/assets/59746974/6e2d53a3-5e03-449e-908a-2d43ff6116da)
 Here is a map of the difference between a given blaze taken before and after the reset.  The difference is not 0 and increases towards the K-band, where the blazes have a shape which is less symetrical.  I understand that the polynomial involved into the blaze flux model is only up to 3rd order, but maybe the non-orthogonality between terms is the culprit for differences we see from one reduction to the next?

**Comment by larnoldgithub** [comment]
Differences are low frenquecies, and likely to be described well with a polynomial up to degree 3.

**Comment by larnoldgithub** [comment]
What I'll do today is another reset on apikipiha and run again the mini_run2, to verify that what I get is consistent.

**Comment by larnoldgithub** [comment]
The results after this second apero-reset suggest that a given machine finds always the same fit for a the blaze. But what about different machines ?   Maybe Neil you could check for any differences in the blazes you got from the different reductions in marseille, toulouse, montreal and cfht ?

**Comment by njcuk9999** [comment]
Yer I'll try checking these tomorrow. Correct me if I'm wrong though but your two results are from the same machine (before and after reset?) so do we have any idea what the difference was?

**Comment by njcuk9999** [comment]
So I went through all 7 machines. And the difference is tiny.
The fluxes we are talking for the blaze are ~80000 to 100000.

The biggest difference I see is on order 10 with a max diff of 0.6 at the edge of the blaze (around a flux of 5000)

Seems to only be cfht out this far on this particular order and all others agree much better:
![image](https://github.com/njcuk9999/apero-drs/assets/6008608/75c64c73-172f-4b1b-a869-103fa2972e85)

All other orders look like this with all not agreeing exactly but all agreeing quite well (within 0.03 flux on 60000)
![image](https://github.com/njcuk9999/apero-drs/assets/6008608/083c9b04-cfc6-4320-a133-2727c60dc54a)

**Comment by njcuk9999** [comment]
So yer most blazes are at numerical levels (fractionally ~10^-7) with the largest deviation being fractional = 10^-5 on order 10.

**Comment by njcuk9999** [comment]
This is with v0.7.288  with the four machines in Montreal:

![image](https://github.com/njcuk9999/apero-drs/assets/6008608/7c125b57-cb6a-4eae-9787-fab8b9309a47)

```
rms of difference spirou@rali-cook@jupiter = 0.15466742511421885 m/s
rms of difference spirou@rali-cook@nb19 = 0.016303989383939683 m/s
rms of difference spirou@rali-spirou@maestria = 0.044794374568119605 m/s
```

If you can just drop your GL699_GL699 files in here: https://drive.google.com/drive/folders/1XpgLS_mD0-vZXMFIJCblZPoKBsC0xoC1?usp=sharing
I'll add to the comparison

@clairem789 turns out my code needs the .fits files too (sorry)

**Comment by larnoldgithub** [comment]
@njcuk9999 I have uploaded my lbl files to your google drive. Just to let you know.

**Comment by larnoldgithub** [comment]
Comparison between CFHT and NW minidataset .288  @njcuk9999 @clairem789
night 2020-05-14
Gl699  *o_pp.fits:   difference = exactly 0
A difference is visible in blaze, order 10.  Also in tcorr.
![Screenshot 2023-10-03 at 22 45 41](https://github.com/njcuk9999/apero-drs/assets/59746974/c3294361-42d9-4784-b404-dc214c24a33f)
![Screenshot 2023-10-03 at 22 47 23](https://github.com/njcuk9999/apero-drs/assets/59746974/9873b74d-a02f-462e-ad3b-c2dcb6f020cc)
![Screenshot 2023-10-03 at 22 48 34](https://github.com/njcuk9999/apero-drs/assets/59746974/06c62af9-ad01-4c6b-b3aa-808fdc3154e1)
![Screenshot 2023-10-03 at 22 49 18](https://github.com/njcuk9999/apero-drs/assets/59746974/887416c4-04b3-4170-b4c3-71a36fa0bdf7)

**Comment by njcuk9999** [comment]
This isn't really a good way to plot this. You need to make the difference fractional or relative to the flux in one of the images.

Instead of a direct difference you need difference/mean value of the two images at each pixel... you'll see its tiny (at the numerical accuracy level). The values of the blaze are huge compared to a value of "10".

**Comment by njcuk9999** [comment]
May be fixed at least to ~4cm/s (pending NewWorlds @clairem789 )

![Image](https://github.com/user-attachments/assets/eab51aac-12d2-4b9b-befc-912fb2c403c8)

```
rms of difference cook@jupiter-spirou@rali = 0.03450295263957454 m/s
rms of difference cook@jupiter-cook@nb19 = 0.044254861303022294 m/s
rms of difference cook@jupiter-spirou@maestria = 0.038549640569937114 m/s
rms of difference cook@jupiter-lam = 0.03423387228007862 m/s
rms of difference cook@jupiter-cfht = 0.03906898180215942 m/s
rms of difference cook@jupiter-cfht@canfar = 0.0343671241526376 m/s
```
